backend/src/openarchiefbeheer/destruction/tests/test_endpoints.py

```diff
@@ -468,7 +469,37 @@ def test_cannot_mark_as_finally_if_not_internally_reviewed(self):
 
         self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
 
-    def test_mark_as_final(self):
+    def test_cannot_mark_as_final_if_posted_user_is_not_archivist(self):
+        record_manager = UserFactory.create(
+            username="record_manager", role__can_start_destruction=True
+        )
+        internal_reviewer = UserFactory.create(
+            username="archivist", role__can_review_final_list=False
+        )
+        destruction_list = DestructionListFactory.create(
+            name="A test list",
+            author=record_manager,
+            contains_sensitive_info=True,
+            status=ListStatus.internally_reviewed,
+        )
+
+        self.client.force_authenticate(user=record_manager)
+        endpoint = reverse(
+            "api:destructionlist-make-final", kwargs={"uuid": destruction_list.uuid}
+        )
+        response = self.client.post(
+            endpoint,
+            data={"user": internal_reviewer.pk},
+            format="json",
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
```

Comment:
Should this be a 400 or 403? Is the request wrong or the operation not permitted? Kinda both :P Please double check but I'll leave it up to you for the definitive decision.

Reply:
I was also unsure initially, but decided that 400 made more sense because here it is the record manager making the request (so they have permission) but they are adding a user without archivist role, so the request is "wrong" :thinking: